spyder/app/utils.py

```diff
@@ -358,6 +367,16 @@ def create_window(WindowClass, app, splash, options, args):
     main.show()
     main.post_visible_setup()
 
+    # Add a reference to the main window so it can be accessed from anywhere.
+    #
+    # Notes
+    # -----
+    # * This should be used to get main window properties (such as width,
+    #   height or position) that other widgets can need to position relative to
+    #   it.
+    # * **DO NOT** use it to access other plugins functionality through it.
+    app.main_window = main
```

Comment:
Rather than giving access to the whole main window shouldn't be created specific accessors? So, leaving this as `app._main_window = main` and creating a set of functions to get the values needed so `get_mainwindow_width`, `get_mainwindow_height`, `get_mainwindow_position`, etc?

Reply:
That's a very good idea! Thanks @dalthviz for the suggestion 👍🏽 

I'll implement it before merging.